### PR TITLE
Feature/quickstart

### PR DESCRIPTION
**Pull request by prashantmital**
Quickstart script is now tested on the following configs:

- OSX 10.10 + Anaconda + MacPorts
- OSX 10.10 + Miniconda + MacPorts
- OSX 10.10 + Anaconda + HomeBrew
- OSX 10.10 + Miniconda + HomeBrew

### CODE REVIEW

**Comment by markkness** [comment]
Nice example!
I do notice, that the LU decomposition example notebook, seems to show an error, at In[17], evaluating d_array[:1, :1]. Since I have not been following this closely, I'll leave it to you all to check if this is correct or not.

**Comment by prashantmital** [comment]
@markkness I recommend following the doc/ge-example branch for keeping a closer eye on the example. I just removed the `ipynb` from this branch in fact as it should have its own commit. I will open a PR from that branch so that you can have a look. 
Also, the LU example does indeed error because slice indexing is not implemented for cyclic distributed arrays. I am trying to add support, but time is limited...

File: docs/quickstart-readme.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+================================================================================
+This script will setup a new conda environment with DistArray. Depending on the
+prior availability of dependencies and system hardware, this process could take
+anywhere from less than a minute upto a few hours. If installation fails,
+delete the created conda environment [conda env remove -n <env-name>]
+and re-run this script.
+
+Prerequisites for using DistArray quickstart:
+- A working conda installation (Anaconda/Miniconda)
+- OSX only: a working package manager (MacPorts/HomeBrew)
```
Comment (bgrant) [inline docs/quickstart-readme.txt]: You should mention that this script *only* works on OS X and Linux (not Windows).

Comment (bgrant) [inline docs/quickstart-readme.txt]: And that you only use MacPorts/Homebrew to install MPI.

**Comment by bgrant** [comment]
Looks good @prashantmital !  Just a couple of comments about the README above.  I'll have to set up OS X on Travis to test it.

**Comment by bgrant** [comment]
Related to #534

**Comment by prashantmital** [comment]
Modified the `quickstart` as needed.

File: docs/quickstart-readme.txt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+[conda env remove -n <env-name>] and re-run the script.
+
+Prerequisites for using DistArray quickstart are:
+- A working conda installation (Anaconda/Miniconda)
+                                      
+Additionally, OSX users will need:
+- A working MPI distribution that provides the 'mpicc' compiler wrapper
+                                      OR
+- A working copy of HomeBrew or MacPorts to install MPI (MacPorts users will
+  need to supply ROOT password when prompted)
```
Comment (bgrant) [inline docs/quickstart-readme.txt]: Not "ROOT password" necessarily- their own account password, for sudo.

Comment (prashantmital) [inline docs/quickstart-readme.txt]: I'll just get rid of the phrasing and say MacPorts users will need SUDO privileges

Comment (bgrant) [inline docs/quickstart-readme.txt]: Great.

**Comment by bgrant** [comment]
Working on testing this on TravisCI now.

A request:  could we have a `-y | --yes` option that skips all the user prompts?

File: quickstart [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    echo "-n | --name <arg>     set name of created conda environment to <arg> DEFAULT distarray"
+    echo "-d | --develop        perform a development install of distarray"
+    echo "-t | --test           test distarray after installation"
+}
+
+while [[ "$1" != "" ]]; do
+    case $1 in
+	-n | --name )
+                shift
+                CONDA_ENV=$1
```
Comment (bgrant) [inline quickstart]: I've noticed there are some tabs in here.  Can we use all spaces?

Comment (prashantmital) [inline quickstart]: Ok.

**Comment by prashantmital** [comment]
Will add the `-y` option ASAP.

**Comment by prashantmital** [comment]
Added -y option. 
Updated user message.
Replaced tabs with spaces.
Looks like this is safe to merge.

**Pull request merged**